### ISSUE TITLE
Questions in the Learning Process

### ISSUE THREAD

**Issue by ChiwoongLEE** [nipponjo/deepfillv2-pytorch#15]
Hello.

I'm trying to learn an image with an image size (720, 1280, 3), and I've confirmed that the initial ether is 1,000,000.

Do we usually proceed with the learning until the iter reaches 1,000,000?
Currently (mine)
iter=180000,
d_loss=1.0000
g_loss=-0.2722
ae_loss=0.0576
ae_loss1=0.0150
ae_loss2=0.0425.

I wonder if it is right to converge to 1 for d_loss.

**Comment by nipponjo** [comment]
Hello, it is hard to know exactly when to stop, so I would mostly look at the results. In the paper, the learning rate is not lowered and I think it is difficult to predict if and when it would make sense.
The the d_loss (hinge loss) is defined as:
```python
hinge_pos = torch.mean(torch.relu(1-pos))
hinge_neg = torch.mean(torch.relu(1+neg))
d_loss = 0.5*hinge_pos + 0.5*hinge_neg 
```
A value of 1 usually means, that the discriminator returns a value around 0 for the original and inpainted image.

**Comment by rafaelvareto** [comment]
I have a technical question regarding random noises commonly added to GANs: 

**Can I assume that the context layer (`ones_x`, the one used for free-form inpainting) somehow replaces the network's requirement for random values?** 

If I'm right, adding random values to `ones_x`, solely to the regions I want to inpaint, I may generate different reconstructions and, therefore, distinct images.
